flight/Modules/Actuator/triflight.c

```diff
@@ -0,0 +1,1018 @@
+                                      float motor_direction_change_angle)
+{
+	float max_phase_shift;
+
+	if (((servo_angle > servo_setpoint_angle) && (servo_angle >= (motor_direction_change_angle + motor_acceleration_delay_angle))) ||
+	    ((servo_angle < servo_setpoint_angle) && (servo_angle <= (motor_direction_change_angle - motor_acceleration_delay_angle))))
+	{
+		// Motor is braking
+		max_phase_shift = fabsf(servo_angle - motor_direction_change_angle) >= motor_deceleration_delay_angle ?
+		                        motor_deceleration_delay_angle:
+		                        fabsf(servo_angle - motor_direction_change_angle);
+	}
+	else
+	{
+		// Motor is accelerating
+		max_phase_shift = motor_acceleration_delay_angle;
+	}
+
+	return max_phase_shift;
+}
+
+/**
+ *
+ *
+ */
+float triGetMotorCorrection(ActuatorSettingsData  *actuatorSettings,
+                            TriflightSettingsData *triflightSettings,
+                            TriflightStatusData   *triflightStatus,
+                            uint16_t servo_value,
+                            float throttle)
+{
+	// Adjust tail motor speed based on servo angle. Check how much to adjust speed from pitch force curve based on servo angle.
+	// Take motor speed up lag into account by shifting the phase of the curve
+	// Not taking into account the motor braking lag (yet)
+
+	float servo_setpoint_angle = getServoAngle(actuatorSettings, triflightSettings, triflightStatus, servo_value); // commanded servo angle
+
+	float max_phase_shift = getPitchCorrectionMaxPhaseShift(triflightStatus->ServoAngle,
+	                                                        servo_setpoint_angle,
+	                                                        tail_motor_acceleration_delay_angle,
+	                                                        tail_motor_deceleration_delay_angle,
+	                                                        tail_motor_pitch_zero_angle);
+
+	float angle_diff = servo_setpoint_angle - triflightStatus->ServoAngle;
+
+	if (fabsf(angle_diff) > max_phase_shift)
+	{
+		angle_diff = max_phase_shift * angle_diff / fabsf(angle_diff);
+	}
+
+	float future_servo_angle = bound_min_max(triflightStatus->ServoAngle + angle_diff,
+	                                         TRI_TAIL_SERVO_ANGLE_MID - triflightSettings->ServoMaxAngle,
+	                                         TRI_TAIL_SERVO_ANGLE_MID + triflightSettings->ServoMaxAngle);
+
+	float correction = (throttle * getPitchCorrectionAtTailAngle(DEG2RAD * future_servo_angle, triflightSettings->MotorThrustFactor)) - throttle;
+
+	return correction;
+}
+
+///////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////
+
+#if !defined(ARCH_POSIX) && !defined(ARCH_WIN32)
```

Comment:
Including this part in the sim caused the sim build to fail, from the ADC use if I recall correctly.  I'll look into it.  Not sure it buys a whole lot in the sim as the values can be set manually, and this is probably fine for the sim since there is no "real" servo either.